trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/CatalogConnectorManager.java

```diff
@@ -146,55 +149,75 @@ public void loadCatalogs(GravitinoMetalake metalake) {
     Set<String> catalogNameStrings =
         Arrays.stream(catalogNames).map(NameIdentifier::toString).collect(Collectors.toSet());
 
-    catalogConnectors.entrySet().stream()
-        .filter(
-            entry ->
-                !catalogNameStrings.contains(entry.getKey())
-                    &&
-                    // Skip the catalog doesn't belong to this metalake.
-                    entry.getValue().getMetalake().name().equals(metalake.name()))
-        .forEach(
-            (entry) -> {
-              catalogInjector.removeCatalogConnector(entry.getKey());
-              catalogConnectors.remove(entry.getKey());
-              LOG.info(
-                  "Remove catalog '{}' in metalake {} successfully.",
-                  entry.getKey(),
-                  metalake.name());
-            });
+    for (Map.Entry<String, CatalogConnectorContext> entry : catalogConnectors.entrySet()) {
+      if (!catalogNameStrings.contains(entry.getKey())
+          &&
+          // Skip the catalog doesn't belong to this metalake.
+          entry.getValue().getMetalake().name().equals(metalake.name())) {
```

Comment:
I'm not sure if this line is necessary. Is it possible that their metalake name is not the same?

Reply:
All the catalogs in different metalakes are managed by CatalogConnectorManager. We should remove the catalog that belongs to this meta lake.

Reply:
Do you mean `different metalakes`,  I remember we can only set one metalake in the configuration file, is it because we have supported adding metalake by the procedure?

Reply:
We need to compare the catalog name. 
if the context in this function:
```
catalogNameStrings = {"test.c1", "test.c2"}
catalogConnectors = {"test.c1", "test.c2", "test.c3", "test1.c4"}
```
If we don't compare the metalake name, the "test.c3", "test1.c4" will dropped.

Reply:
OK